store/v2/root/snapshot_test.go

```diff
@@ -0,0 +1,257 @@
+		}
+	}
+
+	store.Commit()
+	return store
+}
+
+func newMultiStore(t *testing.T, db dbm.DBConnection, stores int) *Store {
+	cfg := multiStoreConfig(t, stores)
+	store, err := NewStore(db, cfg)
+	require.NoError(t, err)
+	return store
+}
+
+func TestMultistoreSnapshot_Errors(t *testing.T) {
+	store := newMultiStoreWithBasicData(t, memdb.NewDB(), 4)
+	testcases := map[string]struct {
+		height     uint64
+		format     uint32
+		expectType error
+	}{
+		"0 height":       {0, snapshottypes.CurrentFormat, nil},
+		"0 format":       {1, 0, snapshottypes.ErrUnknownFormat},
+		"unknown format": {1, 9, snapshottypes.ErrUnknownFormat},
+	}
+	for name, tc := range testcases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			_, err := store.Snapshot(tc.height, tc.format)
+			require.Error(t, err)
+			if tc.expectType != nil {
+				assert.True(t, errors.Is(err, tc.expectType))
+			}
+		})
+	}
+}
+
+func TestMultistoreRestore_Errors(t *testing.T) {
+	store := newMultiStoreWithBasicData(t, memdb.NewDB(), 4)
+	testcases := map[string]struct {
+		height     uint64
+		format     uint32
+		expectType error
+	}{
+		"0 height":       {0, snapshottypes.CurrentFormat, nil},
+		"0 format":       {1, 0, snapshottypes.ErrUnknownFormat},
+		"unknown format": {1, 9, snapshottypes.ErrUnknownFormat},
+	}
+	for name, tc := range testcases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			err := store.Restore(tc.height, tc.format, nil, nil)
+			require.Error(t, err)
+			if tc.expectType != nil {
+				assert.True(t, errors.Is(err, tc.expectType))
+			}
+		})
+	}
+}
+
+func TestMultistoreSnapshot_Checksum(t *testing.T) {
+	store := newMultiStoreWithGeneratedData(t, memdb.NewDB(), 5, 10000)
+	version := uint64(store.LastCommitID().Version)
+
+	testcases := []struct {
+		format      uint32
+		chunkHashes []string
+	}{
+		{1, []string{
+			"28b9dd52156e7c46f42d6c2b390350be3c635a54446f6a6a553e1a6ecca5efca",
+			"8c32e05f312cf2dee6b7d2bdb41e1a2bb2372697f25504e676af1718245d8b63",
+			"05dfef0e32c34ef3900300f9de51f228d7fb204fa8f4e4d0d1529f083d122029",
+			"77d30aeeb427b0bdcedf3639adde1e822c15233d652782e171125280875aa492",
+			"c00c3801da889ea4370f0e647ffe1e291bd47f500e2a7269611eb4cc198b993f",
+			"3af4440d732225317644fa814dd8c0fb52adb7bf9046631af092af2c8cf9b512",
+		}},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(fmt.Sprintf("Format %v", tc.format), func(t *testing.T) {
+			chunks, err := store.Snapshot(version, tc.format)
+			require.NoError(t, err)
+			hashes := []string{}
+			hasher := sha256.New()
+			for chunk := range chunks {
+				hasher.Reset()
+				_, err := io.Copy(hasher, chunk)
+				require.NoError(t, err)
+				hashes = append(hashes, hex.EncodeToString(hasher.Sum(nil)))
+			}
+			assert.Equal(t, tc.chunkHashes, hashes, "Snapshot output for format %v has changed", tc.format)
+		})
+	}
+}
+
+func TestMultistoreSnapshotRestore(t *testing.T) {
+	source := newMultiStoreWithBasicData(t, memdb.NewDB(), 4)
+	target := newMultiStore(t, memdb.NewDB(), 4)
```

Comment:
```suggestion
	target := newMultiStore(t, memdb.NewDB(), 0)
```
this should only work with a new store with empty schema - let's add a check + test for that too

Reply:
ok